pgjdbc/src/main/java/org/postgresql/core/v3/ConnectionFactoryImpl.java

```diff
@@ -587,50 +592,56 @@ private PGStream enableSSL(PGStream pgStream, SslMode sslMode, Properties info,
     }
 
     pgStream.setNetworkTimeout(sslTimeout);
-    // Send SSL request packet
-    pgStream.sendInteger4(8);
-    pgStream.sendInteger2(1234);
-    pgStream.sendInteger2(5679);
-    pgStream.flush();
-
-    // Now get the response from the backend, one of N, E, S.
-    int beresp = pgStream.receiveChar();
-    pgStream.setNetworkTimeout(currentTimeout);
-
-    switch (beresp) {
-      case 'E':
-        LOGGER.log(Level.FINEST, " <=BE SSLError");
-
-        // Server doesn't even know about the SSL handshake protocol
-        if (sslMode.requireEncryption()) {
-          throw new PSQLException(GT.tr("The server does not support SSL."),
-              PSQLState.CONNECTION_REJECTED);
-        }
+    if (!sslNegotiation.startsWith("direct")) {
+
+      // Send SSL request packet
+      pgStream.sendInteger4(8);
```

Comment:
Please flip the order of branches like. It would make code easier to follow, and it would reduce the diff.

```java
if ("direct".equals(sslNegotiation)) {
  MakeSSL.convert(pgStream,info, true);
  return pgStream;
}
...
here follows unmodified SSL code
```